File: Example/FrameWorkDemo/ZallData/ZARootViewController.m
//
//  ZARootViewController.m
//  ZallData
//
//  Created by guo on 2022/1/24.
//  Copyright © 2022 Zall Data Co., Ltd. All rights reserved.
//

#import "ZARootViewController.h"
#import "ZASDKAction.h"


@interface ZARootViewController ()

@property (nonatomic, strong) ZASDKAction *actionModel;


@end

@implementation ZARootViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
}
- (ZASDKAction *)actionModel{
    if (!_actionModel) {
        _actionModel = [[ZASDKAction alloc] init];
    }
    return _actionModel;
}
 
#pragma mark - TableViewDataSource
 

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return self.actionModel.count;
}

-(CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section{
    return 0;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    UITableViewCell * cell =[tableView dequeueReusableCellWithIdentifier:@"cellid" forIndexPath:indexPath];
    cell.textLabel.text = [self.actionModel cellForWithRow:indexPath.row];
    return cell;
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    __weak __typeof(self)weakSelf = self;
    [self.actionModel cellForSelecttWithRow:indexPath.row withBlcok:^(ZASDKAction * _Nonnull action) {
        ZARootViewController * vc = (ZARootViewController *)rootViewController();
        vc.actionModel = action;
        vc.title = [weakSelf.actionModel cellForWithRow:indexPath.row];
        [weakSelf.navigationController showViewController:vc sender:nil];
    }];
}

@end

File: Example/ZallDataSwift/ZARootViewController.swift
//
//  ZARootViewController.swift
//  ZallDataSwift
//
//  Created by Mac on 2022/1/26.
//  Copyright © 2022 Zall Data Co., Ltd. All rights reserved.
//

import UIKit


func rootViewController() -> UIViewController {
    return UIStoryboard.init(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "rootViewController")
}

class ZARootViewController: UIViewController {
    var dataList = ZASDKAction.sdkActionWithDataList(nil)
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Do any additional setup after loading the view.
        self.title = "Track"
    }
}

extension ZARootViewController:UITableViewDelegate,UITableViewDataSource{
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        dataList.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cellid", for: indexPath)
        cell.textLabel?.text = dataList.cellForWithRow(indexPath.row)
        return cell;
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        dataList.cellForSelectWithRow(indexPath.row) { [weak self] action in
            let vc:ZARootViewController = rootViewController() as! ZARootViewController
            vc.dataList = action
            vc.title = self?.dataList.cellForWithRow(indexPath.row)
            self?.navigationController?.show(vc, sender: nil)
            
        }
    }
    
   
    
}
